x/provider/types/msgs_test.go

```diff
@@ -0,0 +1,168 @@
+package types
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pkg/errors"
+	"github.com/tendermint/tendermint/libs/kv"
+)
+
+func TestConfigPath(t *testing.T) {
+	tests := []struct {
+		path   string
+		expErr error
+	}{
+		{
+			path:   "/home/ropes/go/src/github.com/ovrclk/akash/_run/kube/provider.yml",
```

Comment:
this can just be `"/foo"`

Also try:
| value | error? |
|---|---|
|`localhost`|true|
|`localhost/foo`|true|
|`localhost:80`|true|
|`localhost:80/foo`|true|
|`127.0.0.1`|true|
|`127.0.0.1/foo`|true|
|`127.0.0.1:80`|true|
|`127.0.0.1:80/foo`|true|
|`http://localhost`|false|
|`http://localhost/foo`|false|
|`http://localhost:80`|false|
|`http://localhost:80/foo`|false|
|`https://localhost:80`|false|
|`https://localhost:80/foo`|false|

Probably need to audit how this is used/parsed and make sure the validation makes sense.  I think that allowing schemes other than `http(s)` probably doesn't make sense.  Can be done later.

Reply:
might as well throw some ipv6 failing tests in there as well so we don't have to deal with it.